### ISSUE TITLE
Publish to brew/Docker/SDKMAN! via JReleaser

### ISSUE THREAD

**Issue by helpermethod** [flix/flix#4561]
Hi!

First of all, great project, very interesting language!

Now, while giving Flix a shot, I discovered that there was no easy way to install the lastest Flix version apart from building Flix locally. I think it would be great if people could gave it a try by installing it via one of the many popular package managers, like [SDKMAN!](https://sdkman.io/), [Homebrew](https://brew.sh/) or even Docker.

Wdyt? I could lend a hand in integrating this into the build, and I guess @aalmiray would also more than happy to lend a hand 😄 .

**Comment by magnus-madsen** [comment]
Hello @helpermethod and welcome to our little community :) 

That sounds interesting. Not being familiar with these, what would that entail?

For example, I am soon preparing a new release. Usually this means to (manually):

- Create PR with changelog.
- Merge PR, build jar, and upload to GitHub.
- VSCode installations will then automatically grab the latest JAR. 

I assume this is a supplement to that?

**Comment by helpermethod** [comment]
Hi @magnus-madsen,

there are a lot of things possible with GitHub Actions and JReleaser, among them

* automatic release notes generation
* automatic upload of artifacts to GitHub
* announcement of new releases via Twitter
* distribution via Homebrew, SDKMAN! and other packagers
* early-access releases for the more brave users ;)

But I guess as a first step it would make to convert your manual local release process to a manual GH actions based one, and only using JReleaser for the Changelog and Release creation. What do you think?

BTW, how do you do the update the changelog? JReleaser is able to generate the changelog from the commit history, with the possibility to categorize types of changes (dependency updates, features, bugfixes etc.).

Here's an example how a typical JReleaser release looks like https://github.com/jreleaser/jreleaser/releases/tag/v1.2.0

And here's a list of Packagers, which are JReleaser's ways to distribute artifacts.

**Comment by helpermethod** [comment]
I would gladly assist in creating such a workflow!

**Comment by magnus-madsen** [comment]
This looks interesting. I'd like to start with the GitHub release workflow!

- We usually write our own change log because there are many changes, but most our not relevant to end-users. Is it possible to have such a change log at the top and then the generated change log below?

- To prepare a new release, we have to bump the version number in `Version.scala`, run `gradlew jar` and then put the generated JAR on a GitHub release. 

If that could be automated it would be cool!

**Comment by aalmiray** [comment]
Yes, it's possible to have custom release notes. The release notes for JReleaser itself make use of this feature, see https://github.com/jreleaser/jreleaser/blob/main/src/jreleaser/changelog.tpl

You can update the changelog template at any time before a release is made. Alternatively, you may edit the release notes once a release is posted.

**Comment by helpermethod** [comment]
> This looks interesting. I'd like to start with the GitHub release workflow!
> 
> - We usually write our own change log because there are many changes, but most our not relevant to end-users. Is it possible to have such a change log at the top and then the generated change log below?
> 
> - To prepare a new release, we have to bump the version number in `Version.scala`, run `gradlew jar` and then put the generated JAR on a GitHub release. 
> 
> If that could be automated it would be cool! 

I'll see if I can hack something together 😁

**Comment by helpermethod** [comment]
@magnus-madsen So I guess a good first step would be to refactor `Version.scala` to allow passing in the release version at build time. A typical way to do this is:

* let `build.gradle` accept a version property
* add this version to `MANIFEST.MF` via the `Implementation-Version` attribute
* during runtime, read this version from `MANIFEST.MF`

That way, you can trigger a manual build and pass in the release version (in a GitHub workflow, you can pass in properties via `workflow_dispatch`).

Here's a first draft: https://github.com/flix/flix/pull/4578

**Comment by magnus-madsen** [comment]
Sounds approximately reasonable 👍
